src/ShallowTraversal.js

```diff
@@ -27,6 +27,9 @@ export function childrenOfNode(node) {
 
 export function hasClassName(node, className) {
   let classes = propsOfNode(node).className || '';
+  if (node && typeof node.type === 'function' && typeof classes !== 'string') {
```

Comment:
Can we just do the `typeof` check instead of all these checks on `node`? No matter what, if `classes` is not a string we should exit early since the next operation is a string operation.